extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -80,47 +81,42 @@ private CosmosDBInputAttribute GetBindingDataContent(ModelBindingData bindingDat
             Type _ when targetType == typeof(CosmosClient) => CreateCosmosClient<CosmosClient>(cosmosAttribute),
             Type _ when targetType == typeof(Database) => CreateCosmosClient<Database>(cosmosAttribute),
             Type _ when targetType == typeof(Container) => CreateCosmosClient<Container>(cosmosAttribute),
-            _ => await CreateTargetObjectAsync(targetType, cosmosAttribute)
+            _ => ConvertDynamic(await CreateTargetObjectAsync(targetType, cosmosAttribute), targetType)
         };
 
-        private async Task<object> CreateTargetObjectAsync(Type targetType, CosmosDBInputAttribute cosmosAttribute)
+        private object ConvertDynamic(dynamic data, Type targetType)
```

Comment:
Does this work? Since `CreateTargetObjectAsync` returns `Task<object>`

```suggestion
        private object ConvertDynamic(object data, Type targetType)
```